### PR TITLE
Converting old std:: types to cuda::std:: types

### PR DESCRIPTION
**Pull request by cliffburdick**

### CODE REVIEW

File: include/matx/core/tensor_impl.h [hunk 1]
```diff
@@ -209,7 +209,7 @@ class tensor_impl_t {
      *   Data type
      */
     template <typename DescriptorType, std::enable_if_t<is_matx_descriptor_v<typename remove_cvref<DescriptorType>::type>, bool> = true>
-    __MATX_INLINE__ tensor_impl_t(T *const ldata,
+    __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ tensor_impl_t(T *const ldata,
```
Comment (tylera-nvidia) [inline include/matx/core/tensor_impl.h:212]: Did you mean to keep the device qualifier here?

File: include/matx/core/tensor.h [hunk 2]
```diff
@@ -114,7 +114,7 @@ class tensor_t : public detail::tensor_impl_t<T,RANK,Desc> {
    *
    * @param rhs Object to copy from
    */
-  __MATX_HOST__ tensor_t(tensor_t const &rhs) noexcept
+  __MATX_DEVICE__ __MATX_HOST__ tensor_t(tensor_t const &rhs) noexcept
```
Comment (tylera-nvidia) [inline include/matx/core/tensor.h:117]: Did you mean to keep the device qualifier here?

File: include/matx/core/tensor_utils.h [hunk 3]
```diff
@@ -219,7 +219,7 @@ namespace detail {
   }
 
   template <typename T0, typename T1, typename... Tn>
-  constexpr auto matx_max(T0 &&t0, T1 &&t1, Tn &&... tn)
+  constexpr auto  __MATX_HOST__ __MATX_DEVICE__ matx_max(T0 &&t0, T1 &&t1, Tn &&... tn)
```
Comment (tylera-nvidia) [inline include/matx/core/tensor_utils.h:222]: Did you mean to keep the device qualifier here?

File: include/matx/core/type_utils.h [hunk 4]
```diff
@@ -126,7 +126,7 @@ struct is_matx_op_impl<T, std::void_t<typename T::matxop>> : std::true_type {
  * 
  * @tparam T Type to test
  */
-template <typename T> constexpr bool is_matx_op()
+template <typename T> constexpr __MATX_HOST__ __MATX_DEVICE__ bool is_matx_op()
```
Comment (tylera-nvidia) [inline include/matx/core/type_utils.h:129]: Did you mean to keep the device qualifier here?

**Review by tylera-nvidia — APPROVED** [review]

File: include/matx/core/tensor_utils.h [hunk 5]
```diff
@@ -194,7 +194,7 @@ namespace detail {
     if constexpr (is_std_tuple<remove_cvref_t<Tuple>>::value || is_std_array<remove_cvref_t<Tuple>>::value) {
```
[the remark this answers is not in the document]
Comment (cliffburdick) [reply include/matx/core/tensor_utils.h:194]: Good catch

File: include/matx/core/tensor_impl.h (hunk 1 above)
Comment (cliffburdick) [reply include/matx/core/tensor_impl.h:212]: Yes, this was to work around a compiler bug

File: include/matx/core/tensor_utils.h (hunk 3 above)
Comment (cliffburdick) [reply include/matx/core/tensor_utils.h:222]: Yes, for a compiler bug and to get rid of relaxed constexpr

File: include/matx/core/type_utils.h (hunk 4 above)
Comment (cliffburdick) [reply include/matx/core/type_utils.h:129]: Same as others

**Pull request merged**